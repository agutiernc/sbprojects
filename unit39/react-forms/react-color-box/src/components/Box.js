import React from 'react'

const Box = ({ width = 100, bgColor = 'orange', height = 100, id, handleRemove }) => {
  let boxStyle = {
    height: `${height}px`,
    width: `${width}px`,
    backgroundColor: bgColor
  }

  return (
    <div>
      <div id="box" style={boxStyle}>
        <button onClick={() => handleRemove(id)}>Remove Box</button>
      </div>
    </div>
  )
}


export default Box;