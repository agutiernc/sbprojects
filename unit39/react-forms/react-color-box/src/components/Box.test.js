import React from "react";
import { render } from "@testing-library/react";
import Box from './Box'

// smoke test
it('renders without crashing', () => {
  render(<Box />)
})

// snapshot test
it('matches snapshot', () => {
  const { asFragment } = render(<Box />)

  expect(asFragment()).toMatchSnapshot()
})