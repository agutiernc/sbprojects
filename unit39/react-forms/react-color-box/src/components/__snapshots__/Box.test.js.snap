// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      style="height: 100px; width: 100px; background-color: orange;"
    >
      <button>
        Remove Box
      </button>
    </div>
  </div>
</DocumentFragment>
`;
